Microsoft.Azure.Cosmos/src/Resource/ClientContextCore.cs

```diff
@@ -227,18 +228,21 @@ internal override void ValidateResource(string resourceId)
             string operationName,
             RequestOptions requestOptions,
             Func<ITrace, Task<TResult>> task,
+            Func<TResult, OpenTelemetryResponse> openTelemetry = null,
```

Comment:
Don't allow null, it is not a normal scenario and might be hiding a bug if a null does arrive.

```suggestion
            Func<TResult, OpenTelemetryResponse> openTelemetry,
```

Reply:
As per our offline discussion created this issue : https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3212